synapse/handlers/message.py

```diff
@@ -953,18 +953,13 @@ async def create_new_client_event(
             depth=depth,
         )
 
-        old_state = None
+        context = await self.state.compute_event_context(event)
```

Comment:
computing event context isn't free (eg, it creates state groups in the database). We shouldn't do it if we're about to throw it away.